src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -112,7 +119,50 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
             {
                 return null;
             }
+            if (reEnterPassword)
```

Comment:
Above we try to read user name in a cycle until we get valid name or Crtl-C (null).
I'd expect the same behavior for password too.